File: Issue solutions/issue12.cpp
//Count the Number of Set bits in a Binary Number
//This Program is solved using Brian Kernighan’s Algorithm
//Follow up link - https://iq.opengenus.org/brian-kernighan-algorithm/
#include <iostream>
using namespace std;
class Count {
public:
	unsigned int countSetBits(int n)
	{
		unsigned int count = 0;
		while (n) {
			n &= (n - 1);
			count++;
		}
		return count;
	}
};


int main()
{
	Count c;
	int i = 9;
	cout << c.countSetBits(i);
	return 0;
}

File: Issue solutions/Issue_12.java
// This program returns the number of set bits in the binary format of the given number. 
// This solution uses the idea of bitwise operations.

import java.io.*;
 
class countSetBits {
   
    public static int countSetBits(int n)
    {
        int count = 0;
        while (n > 0) {
            count += n & 1;
            n >>= 1;
        }
        return count;
    }
 
    // main function
    public static void main(String args[])
    {
        int i = 9;
        System.out.println(countSetBits(i));
    }
}
